gitpod-io/gitpod
# Browser Extension Advertisement

## Description

Best ad of 2024.

<img width="935" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/3a7b3761-0c38-4d80-b661-838b9baea7d7">

Besides being beautiful and cool, it sends the `browser_extension_promotion_interaction` event with the `action` parameter's value being either `chrome_navigation` or `firefox_navigation`. The dismiss functionality is commented out for now, but one can manually dismiss it by executing
```js
localStorage.setItem("browser-extension-banner-dismissed", true)
```

## Related Issue(s)

Fixes EXP-1812

## How to test

Go on the preview environment and visit `/new` (the new workspace page).

- If you have the extension installed, try to enable it by right-clicking on its icon and clicking on enable. After doing that, refresh the page. The popup should disapear.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-extension-ad</li>
	<li><b>🔗 URL</b> - <a href="https://ft-extension-ad.preview.gitpod-dev.com/workspaces" target="_blank">ft-extension-ad.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-extension-ad-gha.24501</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-extension-ad%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>
